tsconfig.json

```diff
@@ -3,7 +3,8 @@
     "types": ["@types/node"],
     "noEmit": true,
     "allowJs": true,
-    "allowImportingTsExtensions": true
+    "allowImportingTsExtensions": true,
+    "lib": ["ESNext"]
```

Comment:
This matches `packages/website` and allows us to use `Object.groupBy`

Reply:
I don't understand why this one went in,  our primary node version for the repo is 20, and this throws at runtime on node 20

Reply:
Please can you open a PR to revert this and remove incompatible code?

Reply:
cc @JoshuaKGoldberg

Reply:
`ESNext` is never going to be what we want, we are always going to want to work based on a specific version of node that isn't the bleeding edge

Reply:
yep! sorry about that, I was matching `website`'s tsconfig. I'll open a PR now

Reply:
Thanks, please could you try updating website to match with `ES2023` then. It shouldn't cause any issues, looks like it was set to ESNext 3 years ago

Reply:
Well of course it _could_ cause type issues if people have added such bleeding edge constructs recently

Reply:
But let's see :D

Reply:
On my TS project references branch it doesn't seem to